lib/wibox/layout/stack.lua

```diff
@@ -74,10 +74,16 @@ function stack:layout(_, width, height)
 
     local h_off, v_off = spacing, spacing
 
-    for _, v in pairs(self._private.widgets) do
+    if self._private.top_only then
```

Comment:
Does anyone know if there is a way to leave this check in the for loop? Having the check outside the loop doesn't look very elegant.

I had to take it out because the loop starts with the last layer, so the last layer would show when `top_only` is set instead of the top layer.

Lua doesn't allow us to set loop counter variables, so we can't do something like:
```diff
        -- Iterate backwards to draw top (index 1) last so that it appears above other layers
        for i = #self._private.widgets, 1, -1 do
+           if self._private.top_only then i = 1 end
            local v = self._private.widgets[i]
            table.insert(result, base.place_widget_at(v, h_off, v_off, width, height))
            h_off, v_off = h_off + self._private.h_offset, v_off + self._private.v_offset
        end
```